hack/deployer/runner/eks.go

```diff
@@ -245,4 +246,35 @@ func (e *EKSDriver) writeAWSCredentials() error {
 	return os.WriteFile(file, []byte(fileContents), 0600)
 }
 
+func (e *EKSDriver) Cleanup() ([]string, error) {
+	if err := e.auth(); err != nil {
+		return nil, err
+	}
+	daysAgo := time.Now().Add(-24 * 3 * time.Hour)
+	e.ctx["Date"] = daysAgo.Format(time.RFC3339)
+	allClustersCmd := `eksctl get cluster -r "{{.Region}}" -o json | jq -r 'map(select(.Name|test("eck-e2e")))| .[].Name'`
+	allClusters, err := exec.NewCommand(allClustersCmd).AsTemplate(e.ctx).OutputList()
+	if err != nil {
+		return nil, err
+	}
+	var deletedClusters []string
+	describeClusterCmd := `aws eks describe-cluster --name "{{.ClusterName}}" --region "{{.Region}}" | jq -r --arg d "{{.Date}}" 'map(select(.cluster.createdAt | . <= $d))|.[].cluster.name'`
+	for _, cluster := range allClusters {
+		e.ctx["ClusterName"] = cluster
+		clustersToDelete, err := exec.NewCommand(describeClusterCmd).AsTemplate(e.ctx).Output()
+		if err != nil {
+			return nil, fmt.Errorf("while describing cluster %s: %w", cluster, err)
+		}
+		if clustersToDelete != "" {
+			deletedClusters = append(deletedClusters, cluster)
+			log.Printf("deleting cluster %s", cluster)
+			if err := e.newCmd("eksctl delete cluster -v 1 --name {{.ClusterName}} --region {{.Region}} --wait").Run(); err != nil {
```

Comment:
I think we should create `func (e *EKSDriver) delete() error {` and reuse it here and in `func (e *EKSDriver) Execute() error` where we also delete the cluster.